editor/src/components/canvas/ui/floating-insert-menu.tsx

```diff
@@ -114,7 +118,26 @@ export function useGetInsertableComponents(
     }
   }, [packageStatus, propertyControlsInfo, projectContents, dependencies, fullPath, insertMenuMode])
 
-  return insertableComponents
+  const scopedVariables = useEditorState(
+    Substores.variablesInScope,
+    (store) =>
+      getVariablesInScope(
+        store.editor.selectedViews[0],
+        projectContents,
+        store.editor.variablesInScope,
+      ),
+    'useGetInsertableComponents scopedVariables',
+  )
+
+  const insertableVariables = React.useMemo(() => {
+    if (fullPath == null) {
+      return []
+    } else {
+      return convertInsertableComponentsToFlatList(convertVariablesToElements(scopedVariables))
+    }
+  }, [fullPath, scopedVariables])
```

Comment:
I _think_ the scopedVariables is not referentially equal so it will always blow this memo, if the performance is a concern we can look into wrapping `getVariablesInScope` in selectors

Reply:
Oh. Does `insertableComponents` dependencies have the same issue? (line 109)?

Reply:
Yes, good catch. Should be wrapped so it won't be re-created every time.